test/functional/wallet_balance.py

```diff
@@ -89,10 +89,9 @@ def run_test(self):
         # Same with minconf=0
         assert_equal(self.nodes[0].getbalance(minconf=0), Decimal('9.99'))
         assert_equal(self.nodes[1].getbalance(minconf=0), Decimal('29.99'))
-        # getbalance with a minconf incorrectly excludes coins that have been spent more recently than the minconf blocks ago
-        # TODO: fix getbalance tracking of coin spentness depth
-        assert_equal(self.nodes[0].getbalance(minconf=1), Decimal('0'))
-        assert_equal(self.nodes[1].getbalance(minconf=1), Decimal('0'))
+        # getbalance with a minconf includes coins that have been spent more recently than the minconf blocks ago
+        assert_equal(self.nodes[0].getbalance(minconf=1), Decimal('50'))
+        assert_equal(self.nodes[1].getbalance(minconf=1), Decimal('50'))
```

Comment:
This is still the incorrect behaviour, I think. The correct values that should be returned here (at least with `dummy="*"`) are 9.99 and -10.01, respectively. This is because sent transactions are always excluded from the balance, but incoming transactions are only included once they have sufficiently confirmed.